### PR TITLE
Credential Proof support

### PR DESCRIPTION
**Pull request by chaoxinhu**
## Summary
- Concisely describe the issue and your solution: Add credential proof support
- Link to existing issue (type "#" to check): will add #59 
- Specify the tests done so far: UT

## Solutions (Optional)
Please describe the details of your solution if it might be too complicated for reviewers.

### CODE REVIEW

File: src/main/java/com/webank/weid/constant/CredentialConstant.java [hunk 1]
```diff
@@ -38,4 +38,20 @@
      */
     public static final String CREDENTIAL_CONTEXT_PORTABLE_JSON_FIELD = "@context";
 
+    /**
+     * The Credential Proof Type Enumerate.
+     */
+    public static enum CredentialProofType {
+        ECDSA("EcdsaSignature");
+
+        private String typeName;
```
Comment (chenhaozx) [inline src/main/java/com/webank/weid/constant/CredentialConstant.java]: 对于无法一眼知晓含义的属性，请加注释

File: src/main/java/com/webank/weid/service/impl/CredentialServiceImpl.java [hunk 2]
```diff
@@ -100,7 +101,9 @@
             result.setId(UUID.randomUUID().toString());
             result.setCptId(args.getCptId());
             result.setIssuer(args.getIssuer());
-            result.setIssuranceDate(DateUtils.getCurrentTimeStamp());
+            Long issuanceDate = DateUtils.convertUtcDateToTimeStamp(
+                DateUtils.convertTimestampToUtc(DateUtils.getCurrentTimeStamp()));
```
Comment (chenhaozx) [inline src/main/java/com/webank/weid/service/impl/CredentialServiceImpl.java]: 取当前时间戳为啥不直接用System.currentTimeMillis()？

Comment (webankadmin) [reply src/main/java/com/webank/weid/service/impl/CredentialServiceImpl.java]: 放到DateUtils里面，这里有个小bug，为了支持离线生成的Credential

Comment (chaoxinhu) [reply src/main/java/com/webank/weid/service/impl/CredentialServiceImpl.java]: 这个拿掉了，后面放到rest分支里面

File: src/main/java/com/webank/weid/protocol/base/CredentialProof.java [hunk 3]
```diff
@@ -0,0 +1,48 @@
...
+     */
+    private String type;
+
+    /**
+     * Required: The create date.
+     */
+    private Long created;
+
+    /**
+     * Required: The create date.
```
Comment (chaoxinhu) [inline src/main/java/com/webank/weid/protocol/base/CredentialProof.java]: creator

File: src/main/java/com/webank/weid/util/CredentialUtils.java [hunk 4]
```diff
@@ -294,13 +368,59 @@ public static ErrorCode isCredentialContentValid(Credential args) {
...
+     * @return Context value in String.
+     */
+    public static String getDefaultCredentialProofType() {
+        return CredentialConstant.CredentialProofType.ECDSA.getTypeName();
+    }
+
+    private static boolean isCredentialProofTypeValid(String type) {
+        // Proof type must be one of the pre-defined types.
+        if (!StringUtils.isEmpty(type)) {
+            for (CredentialProofType types : CredentialConstant.CredentialProofType.values()) {
```
Comment (chenhaozx) [inline src/main/java/com/webank/weid/util/CredentialUtils.java]: types应该是type吧？

File: src/main/java/com/webank/weid/service/impl/CredentialServiceImpl.java [hunk 5]
```diff
@@ -309,11 +306,12 @@ private ErrorCode verifyCptFormat(Integer cptId, Map<String, Object> claim) {
             Credential credential = credentialWrapper.getCredential();
             Map<String, Object> disclosureMap = credentialWrapper.getDisclosure();
             String rawData = CredentialUtils
-                .getCredentialThumbprintWithoutSig(credential, disclosureMap);
+                .getCredentialThumbprintWithoutSignature(credential, disclosureMap);
```
[the remark this answers is not in the document]
Comment (chaoxinhu) [reply src/main/java/com/webank/weid/service/impl/CredentialServiceImpl.java]: done

**Pull request closed without merging**